toolkit/tools/internal/customizationmacros/customizationmacros.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Microsoft Corporation.
+// Licensed under the MIT License.
+
+/*
+customizationmacros package provides functions to add customization macros to a root directory. These macros are saved in the
+root directory's default rpm macros directory. Each customization has a corresponding macro file that is generated as needed.
+*/
+package customizationmacros
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/rpm"
+)
+
+const (
+	// macro files to customize the installer and final images
+	DisableDocsMacroFile         = "macros.installercustomizations_disable_docs"
+	ConfigureRpmLocalesMacroFile = "macros.installercustomizations_customize_locales"
+)
+
+var (
+	// Universal header comments for all customization macro files
+	customizationMacroHeaderComments = []string{
+		"# This macro file was dynamically generated by the Azure Linux Toolkit image generator",
+		"# based on the configuration used at image creation time.",
+		"",
+	}
+	docComments []string = []string{
+		"# This stops documentation files (anything fine in a package which is %%doc) from being installed.",
+		"# To enable documentation files, remove this file, or comment out '%%_excludedocs 1'",
+		"# Any packages which are already installed must be reinstalled for this change to take effect.",
+	}
+	localeComments []string = []string{
+		"# This stops locale files from being installed. %%_install_langs acts as a filter for locales",
+		"# which start with the provides strings. Setting it to an invalid value (ie 'NONE') will",
+		"# prevent any locale files from being installed.",
```

Comment:
"en" isn't actually a default, there can be "en*" locale files. The default is whatever is baked into the binaries. They way the filter works is: For each of "<loc1>:<loc2>:<loc3>..." set, it will include any locale path that looks like: `<loc1>*/`, `<loc2>*/`, `<loc3>*/`. So since no locale will start with "NONE", no locale paths make it past the filter.